### PR TITLE
[tests-only] [full-ci] API test for enforce password config and update the public share to edit permission

### PR DESCRIPTION
**Pull request by PrajwolAmatya**
## Description
This PR adds API test for setting the config `OCIS_SHARING_PUBLIC_WRITEABLE_SHARE_MUST_HAVE_PASSWORD` to true and update the public share to edit permission.

## Related Issue
- Related to issue: https://github.com/owncloud/ocis/issues/6231

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:5]:
```suggestion
  I want to enforce passwords on public links shared with upload, edit, or contribute permission
  So that the password is required to make changes to the shared resources
```

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:11]: let's do the config thing before
```suggestion
    Given the config "OCIS_SHARING_PUBLIC_WRITEABLE_SHARE_MUST_HAVE_PASSWORD" has been set to "true"
    And these users have been created with default attributes and without skeleton files:
      | username |
      | Alice    |
```

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:14]:
```suggestion
  Scenario: user tries to create public link with edit permission without password while enforce password is enabled
```

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:25]:
```suggestion
  Scenario: user creates a public link with edit permission while enforce-password is enabled
```

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature
Comment (SwikritiT) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:5]:
```suggestion
  I want to enforce passwords on public links shared with upload, edit, or contribute permission
  So that unauthorized people won't gain access to the contents in the link
```

Comment (saw-jan) [reply tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:5]: IMO, unauthorized could be misleading here. But I don't have any good sentences to suggest here :smile:

Comment (saw-jan) [reply tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:5]: Maybe `So that the password is required to access the contents in the link`? But the link can be with read-only permission that doesn't require password to be set

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature
Comment (SagarGi) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:14]:
```suggestion
  Scenario: user tries to update a public link to edit permission without a password when enforce-password is enabled
```

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature
Comment (SagarGi) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:25]:
```suggestion
  Scenario: user tries to update a public link to edit permission with a password when enforce-password is enabled
```

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature
Comment (amrita-shrestha) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:33]: don't we get ocs message?

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature
Comment (amrita-shrestha) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:26]:
```suggestion
    Given user "Alice" has uploaded file with content "test file" to "/testfile.txt"
```
this step can go to background

File: tests/acceptance/features/bootstrap/OcisConfigContext.php
Comment (amrita-shrestha) [inline tests/acceptance/features/bootstrap/OcisConfigContext.php:96]: communicate because both PR contain same changes https://github.com/owncloud/ocis/pull/6402/files

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature
Comment (PrajwolAmatya) [reply tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:33]: We get ocs message which only contains `OK`. Should we arrest that message too?

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature
Comment (SwikritiT) [reply tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:5]: I think `So that the password is required to access the contents in the link` is okay because the feature is for `upload, edit, or contribute permission` permission only
`I want to enforce passwords on public links shared with upload, edit, or contribute permission`

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature
Comment (SwikritiT) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:36]: The indent looks weird, also do we have a step like
```feature
 When the public accesses the preview of the file "textfile.txt" from the last shared public link using the sharing API
```
```suggestion
    When the public accesses the preview of the following files from the last shared public link using the sharing API
      | path          |
      | /textfile.txt |
```

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature
Comment (SwikritiT) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:37]: Wouldn't this work here?
```suggestion
    Then the HTTP status code should be "404"
```

**Comment by PrajwolAmatya** [comment]
There is a issue related to this PR. https://github.com/owncloud/ocis/issues/6401
So, currently blocking this PR.

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature
Comment (amrita-shrestha) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:4]:
```suggestion
  I want to enforce passwords on public links 
```

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature
Comment (amrita-shrestha) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:5]:
```suggestion
  So that the password is required to access the contents of the link
```

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature
Comment (amrita-shrestha) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:35]: why we need this step in this scenario mostly we avoid using multiple when.

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature
Comment (amrita-shrestha) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:32]:
```suggestion
    When user "Alice" updates the last public link share using the sharing API with
      | permissions | 3    |
      | password    | 1234 |
      Then the HTTP status code should be "200"
    And the OCS status code should be "100"
```

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature
Comment (saw-jan) [reply tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:4]: not all public link gets password enforced. For now, password is enforced for links with upload,edit and contribute permissions. See https://doc.owncloud.com/ocis/next/deployment/services/s-list/sharing.html

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature
Comment (SagarGi) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:31]: may be this can be changed to given step to avoid multiple when step or else please do it as `When` followed by `And` for multiple when step.

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature
Comment (PrajwolAmatya) [reply tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:35]: Just checking if the public link can be accessed without password. Should I remove or keep this step?

Comment (PrajwolAmatya) [reply tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:35]: I think this step can be removed.

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature
Comment (SagarGi) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:22]: May be 996 is `internal sever error`. May be you need to create an issue for it IMO? @PrajwolAmatya

Comment (saw-jan) [reply tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:22]: There is one https://github.com/owncloud/ocis/issues/6476

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature
Comment (SwikritiT) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:25]: should this be tries to or updates, it's the passing case isn't it?
```suggestion
  Scenario: user updates a public link to edit permission with a password when enforce-password is enabled
```

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature
Comment (SwikritiT) [reply tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:35]: IMO it's okay to have this when the step to be sure that the public doesn't have access to the link without a password.
Do we have a separate scenario where this is tested? If yes then probably not needed here.

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature
Comment (SwikritiT) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:25]: Also, I don't think the `when enforce-password is enabled` is needed because the whole feature about that
```suggestion
  Scenario: user tries to update a public link to edit permission with a password
```

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature
Comment (SwikritiT) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:15]: I don't think the `when enforce-password is enabled` is needed because the whole feature about that
```suggestion
  Scenario: user tries to update a public link to edit permission without a password 
```

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:35]: Add step to download/access the public link using the password. should be already available
```suggestion
    And the public should not be able to download file "/textfile.txt" from inside the last public link shared folder using the new public WebDAV API without a password
    And the public should not be able to download file "/textfile.txt" from inside the last public link shared folder using the new public WebDAV API with password "wrong pass"
    But the public should be able to download file "/textfile.txt" from inside the last public link shared folder using the new public WebDAV API with password "1234"
```

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:28]: expecting correct codes

**Review by saw-jan — COMMENTED** [review]
should wait for the fix: https://github.com/owncloud/ocis/issues/6476

**Comment by 2403905** [comment]
@PrajwolAmatya @saw-jan  I already finished the enforce password in a reva backend for the create /update actions [#6476 ](https://github.com/owncloud/ocis/issues/6476). Before we merge and bump the reva we have to update the tests.
Now I see a lot of failed scenarios.
```text
make test-acceptance-from-core-api \
TEST_SERVER_URL=https://localhost:9200 \
TEST_WITH_GRAPH_API=true \
TEST_OCIS=true \
BEHAT_FEATURE=tests/acceptance/features/coreApiSharePublicLink3/updatePublicLinkShare.feature

.....

46 scenarios (32 passed, 14 failed)
502 steps (447 passed, 14 failed, 41 skipped)
0m48.40s (17.74Mb)
grep: empty (sub)expression
grep: empty (sub)expression

The following tests were skipped because they are tagged @skip:
runsh: Total 0 scenarios (0 passed, 0 failed)
runsh: Total unexpected failed scenarios throughout the test run:
coreApiSharePublicLink3/updatePublicLinkShare.feature:182
coreApiSharePublicLink3/updatePublicLinkShare.feature:183
coreApiSharePublicLink3/updatePublicLinkShare.feature:214
coreApiSharePublicLink3/updatePublicLinkShare.feature:215
coreApiSharePublicLink3/updatePublicLinkShare.feature:246
coreApiSharePublicLink3/updatePublicLinkShare.feature:247
coreApiSharePublicLink3/updatePublicLinkShare.feature:288
coreApiSharePublicLink3/updatePublicLinkShare.feature:289
coreApiSharePublicLink3/updatePublicLinkShare.feature:330
coreApiSharePublicLink3/updatePublicLinkShare.feature:331
coreApiSharePublicLink3/updatePublicLinkShare.feature:352
coreApiSharePublicLink3/updatePublicLinkShare.feature:353
coreApiSharePublicLink3/updatePublicLinkShare.feature:374
coreApiSharePublicLink3/updatePublicLinkShare.feature:375
```

**Comment by 2403905** [comment]
@PrajwolAmatya @saw-jan  The ocis tests are green for #6476  [https://github.com/owncloud/ocis/pull/6564](https://github.com/owncloud/ocis/pull/6564)

**Comment by saw-jan** [comment]
> @PrajwolAmatya @saw-jan The ocis tests are green for #6476 #6564

So now, reva PR should be merged and bumped reva in ocis and then we can change the expected statuscode in this PR then merge.

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature
Comment (SagarGi) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:15]:
```suggestion
  Scenario Outline: create a public link with edit permission without a password when enforce-password is enabled
```

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature
Comment (SagarGi) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:29]:
```suggestion
  Scenario Outline: update a public link to edit permission without a password
```

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature
Comment (SagarGi) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:45]:
```suggestion
  Scenario Outline: update a public link to edit permission with a password
```

**Review by saw-jan — APPROVED** [review]
LGTM

**Review by SwikritiT — APPROVED** [review]
LGTM

**Pull request merged**
